### PR TITLE
Host device map fixes: implement host iterator_at and unify map loops…

### PR DESCRIPTION
**Pull request by liu15**
…; reset size in host clear(); fix loop fuser map macros for host; remove START argument for MAP macros

### CODE REVIEW

File: src/care/host_device_map.h [hunk 1]
```diff
@@ -90,34 +94,52 @@ namespace care {
...
+              }
+           }
+           ++(*m_next_iterator_index);
+           return (*m_iterator)++;
+        }
 
       private:
          // we do a heap allocated map to ensure no deep copies occur during lambda capture
          std::map<key_type, mapped_type> * m_map = nullptr;
+         mutable typename std::map<key_type, mapped_type>::iterator * m_iterator = nullptr;
```
Comment (robinson96) [inline src/care/host_device_map.h:141]: you shouldn't need the mutables since the pointer value itself isn't changing in a lambda context

Comment (liu15) [reply src/care/host_device_map.h]: Ah, yes.  I think I was missing a dereference in an earlier version (which caused the compiler to complain about const-ness).

**Review by robinson96 — APPROVED** [review]
LGTM

**Review by adayton1 — APPROVED** [review]
LGTM

**Pull request merged**